### PR TITLE
Start of helicopter Implementation

### PR DESCRIPTION
**Pull request by tahg**
Mostly Matching, some issues with MxMatrix and/or MxVector4

### CODE REVIEW

File: LEGO1/islepathactor.cpp [hunk 1]
```diff
@@ -11,7 +11,6 @@ IslePathActor::IslePathActor()
   this->m_unk158 = 0;
 }
 
-// OFFSET: LEGO1 0x10002e10
-IslePathActor::~IslePathActor()
-{
-}
+MxResult IslePathActor::InitFromMxDSObject(MxDSObject &p_dsObject) {
```
Comment (foxtacles) [inline LEGO1/islepathactor.cpp:14]: Needs an offset comment, should be at `1001a280`

File: LEGO1/helicopter.cpp [hunk 2]
```diff
@@ -1,13 +1,41 @@
 #include "helicopter.h"
+#include "act3.h"
+#include "legoomni.h"
+#include "legocontrolmanager.h"
+#include "legogamestate.h"
+#include "legoworld.h"
 
-// OFFSET: LEGO1 0x10001e60 STUB
```
Comment (foxtacles) [inline LEGO1/helicopter.cpp:3]: Can you add a `DECOMP_SIZE_ASSERT` for the class? Thanks!

File: LEGO1/act3.h [hunk 3]
```diff
@@ -25,6 +25,10 @@ class Act3 : public LegoWorld
     return !strcmp(name, Act3::ClassName()) || LegoWorld::IsA(name);
   }
 
+  inline void SetUnkown420c(MxEntity *p_entity) { m_unk420c = p_entity; }
+  protected:
+    undefined m_unkf8[0x4114];
```
Comment (foxtacles) [inline LEGO1/act3.h:30]: Same here, `DECOMP_SIZE_ASSERT` would be great. Helps prevent some accidental changes to the structure.

File: LEGO1/helicopter.h [hunk 4]
```diff
@@ -24,6 +25,24 @@ class Helicopter : public IslePathActor
...
+  MxMatrixData m_unk1a8;
+  undefined4 m_unk1f0;
+  MxVector4Data m_unk1f4;
+  MxVector4Data m_unk20c;
+  undefined4 m_unk224;
+  HelicopterState *m_state;
+  MxAtomId m_unk22c;
+
+private:
+  void Helicopter::GetState();
```
Comment (foxtacles) [inline LEGO1/helicopter.h:45]: The class name shouldn't be prepended (`Helicopter::`)

File: LEGO1/islepathactor.h [hunk 5]
```diff
@@ -26,8 +25,20 @@ class IslePathActor : public LegoPathActor
     return !strcmp(name, IslePathActor::ClassName()) || LegoPathActor::IsA(name);
   }
 
+  // OFFSET: LEGO1 0x10002ff0 TEMPLATE
+  // IslePathActor::`scalar deleting destructor'
+  inline virtual ~IslePathActor()
```
Comment (foxtacles) [inline LEGO1/islepathactor.h:30]: same here wrt override

**Comment by tahg** [comment]
Thanks I'll fix these up.

**Review by foxtacles — APPROVED** [review]
Looks good, thank you!

**Pull request merged**